### ISSUE TITLE
Support for Nuki Smart Lock

### ISSUE THREAD

**Issue by kaznad** [bwssytems/ha-bridge#140]
Hi, is it possible to use this habridge to lock/unlock the Nuki Smart lock? [https://nuki.io/en/](url)
They support REST API, so can I use the habridge to send something like, http://10.0.0.2:8080/lockAction?nukiId=1&action=1&token=c8v7b6 to lock/unlock the Nuki Smart lock?

Thanks.

**Comment by bwssytems** [comment]
You could do a manual add currently and that should give you a result. Now, opening up the bridge to be able to unlock the door is a security risk.

**Comment by itsupportcg** [comment]
Hi, thanks for the update. Just wondering, isn't the habridge only accessible on the local network? And the http calls are also local calls, so why is there a security risk?

Thanks.

**Comment by kaznad** [comment]
Hi, thanks for the update. Just wondering, isn't the habridge only accessible on the local network? And the http calls are also local calls, so why is there a security risk?

Thanks.

**Comment by osenbruggen** [comment]
Uhhh you Amazon Echo listens to everyone, not just your voice. If you yell hard enough outside or start a recording you slide under the door the lock will open ;)

**Comment by kaznad** [comment]
> Uhhh you Amazon Echo listens to everyone, not just your voice. If you yell hard enough outside or start a recording you slide under the door the lock will open ;)

I am not sure if this is what bwsystems meant, but actually I dun intend to use it with Echo. I plan to use it with my Harmony Hub, which can be integrated to smartthings. So I can setup automation such as using a presence sensor to unlock the door upon detection or to lock the door when the system is in armed mode. Also beside the door lock, I have a main gate lock which is using traditional lock. :)

If habridge was able to integrate with smartthings directly that would be really useful. I assume there is no way to do it right? I did try finding philips hue on my smartthings hub but it only showed my original philips hue V2 hub.

**Comment by rafale77** [comment]
Well you are choosing to do it in a very complex way. This bridge is initially designed to bridge the Vera to the amazon Echo. Then more devices were added like the Harmony hub and the hue hub. If what you are trying to do is to bridge a lock to the ST, you should just use an ST smart app to go directly to the lock.

**Comment by kaznad** [comment]
> Well you are choosing to do it in a very complex way. This bridge is initially designed to bridge the Vera to the amazon Echo. Then more devices were added like the Harmony hub and the hue hub. If what you are trying to do is to bridge a lock to the ST, you should just use an ST smart app to go directly to the lock.

You are absolutely right, I am doing it in a very complex way. I know it will be easier to use create a device handler in Smartthings, however I do not have the skill to do that. :laughing:  So this is the only way I can integrate it to Smartthings. I guess there will be some community development for nuki soon but for now this is only way I know how to do it.

**Comment by audiofreak9** [comment]
@kaznad, I know you are not planning to use this with an Echo, but just to add a thought to this security idea, I am able to call my home and leave a message on my answering machine that the Echo can hear, so you don't even have to yell from outside, be careful.

**Comment by bwssytems** [comment]
The ha-bridge itself is inherently unsecure since the api is open for anything on the network to be called. I do understand that you are only using it on your local network, but there are people that try to hack into home networks.

So, you should be able to do this. the first thing is to unplug your real philips hue before discovering the ha-bridge with your hub, it seems a lot of systems will get the response from the hue hub first and they do not try to find other ones after that.

**Comment by kaznad** [comment]
Hi, I have tried unplugging the Real Hue Bridge and still Smartthings App does not discover the HA Bridge.

Anyone has added their HA Bridge to Smartthings successfully?

Thanks.

**Comment by bwssytems** [comment]
When you try to have Smartthings discover the HA bridge, have trace upnp turned on in the ha bridge config and then you can see the output if it is working.

**Comment by kaznad** [comment]
This is the log that I can see.

08-17-2016 13:27:55.630	INFO	Traceupnp: isSSDPDiscovery found message to be an M-SEARCH message.	com.bwssystems.HABridge.upnp.UpnpListener
08-17-2016 13:27:55.631	INFO	Traceupnp: isSSDPDiscovery found message to be valid under strict rules - strict: true	com.bwssystems.HABridge.upnp.UpnpListener
08-17-2016 13:27:55.631	INFO	Traceupnp: SSDP packet from 192.168.1.12:10023, body: M-SEARCH * HTTP/1.1
 MX: 4
 MAN: "ssdp:discover"
 HOST:239.255.255.250:1900
 ST: urn:schemas-upnp-org:device:basic:1
 
	com.bwssystems.HABridge.upnp.UpnpListener
08-17-2016 13:27:55.631	INFO	Traceupnp: sendUpnpResponse discovery template with address: 192.168.1.3 and port: 80	com.bwssystems.HABridge.upnp.UpnpListener

The IP of ST hub is showing in the log. " SSDP packet from 192.168.1.12:10023"
But it is not discovering. I have tried restarting the ST hub as well. I also tried changing the port to 80 and it still could not be discovered.
